### ISSUE TITLE
Refined searches

### ISSUE THREAD

**Issue by bgregg** [umts/pvta-multiplatform#43]
Right now, a search query searches the entire object (i.e, if you search '30' and a different route has a route id containing 30, it will also show up). The search function should be refined.

**Comment by akaplo** [comment]
FYI: the [messages endpoint](http://bustracker.pvta.com/infopoint/rest/publicmessages/getcurrentmessages) has an array property called **Routes**.  Simply checking if (the route id of the current route being displayed) is in ``messages.Routes`` should do it.

**Comment by dfaulken** [comment]
It might reduce unexpected results if we just search over the names of things.

**Comment by akaplo** [comment]
What would be the best way to go about this?

**Comment by akaplo** [comment]
Since this issue was opened, the search functions have been indirectly refined (by the fact that we strip the objects of their properties we don't need in (I think) all cases), I propose marking this as resolved.

Objections?

**Comment by akaplo** [comment]
Hearing none.

**Comment by sherson** [comment]
Reopening because during today's demo with PVTA, the case was made for searching by stop ID. 

This didn't specifically come up during the demo, but if I search for a stop number, I get results for all stops that contain that number.

**Comment by akaplo** [comment]
So I'm not sure if there's a way to do exactly what we want.

For example, if I search for stop `80`, it's also going to display stop `800`, since `80` is of course contained within `800`.  I don't think that the library we use has a way to check whether it's exactly 80 or not.  Whatever we do would be applied to entire objects, thus, if we only returned stops with IDs that match exactly what we're looking for, we'd also have to return stops with **names** that match exactly what we're looking for.

For example, if we were to change the library's search function to operate in this manner `arnold` would return nothing. `arnold hous` would return nothing.  `arnold house` would return the intended value, but that's a little much to ask of people.

**Comment by werebus** [comment]
How much of a terrible hack is it to change the begins-with/partial-match behaviour depending on whether the user started with a digit? Of course that makes these stops a little harder to find:
- 3300 Main Street (1007)
- 1 Stop Plaza (1546)
- 2133 or 2110 Boston Rd (Flag stop - signal driver)
- 2443 or 2440 Boston Rd (Flag stop - signal driver)
- 806 Stoney Hill (Flag stop - signal driver) (5031)
- 90 Mill St (6247)
- 46 Main St. (666)
- 54 Industrial Drive (9012)
- 50 Wason Ave (9129)
- 100 Wasson Ave (9202)
- 333 Western Ave (937)
- 3550 Main St (967)
- 3550 Main Street (978)
- 3350 Main St (994)

I'm not sure that's what we _should_ do, though. It just seems like the most direct approach to the problem that "people" seem to want two different types of searching in one interface.

**Comment by akaplo** [comment]
So, after looking further into our code and [our searching library's docs](https://github.com/djett41/ionic-filter-bar), I have come up with a few compromises, and I'm interested to hear what you all think (@sherson this is for you too).

For searching routes, I've narrowed it down so that it returns a route **only** if the `ShortName` or `LongName` include the text we're searching for.

For searching stops, I've gotten a little clever:
I define a regex that determines whether the entire input is comprised of digits.

``` javascript
/^\d+$/
```

If so, add a stop to the results if the `StopId` is completely equal to the input (we could do a `contains`, but that returns a ton of results).

If not, that means our input contains characters in the standard 26-letter alphabet, so we can assume they're looking for a name.  Add a stop to the results iff its name includes our input query.

**Comment by akaplo** [comment]
For @werebus's examples above, this method works with 1 caveat (which I'll show by example below):

If you type in the stop name exactly (aka "54 Industrial Drive"), you won't see results for the correct stop until you've started entering characters.  "54" will return "Village Park (N)", because that's the stop with an ID of 54.  However, once you begin adding characters ("54 I"), you'll get a single result: "54 Industrial Drive"

Basically the tradeoff is whether or not someone gives up trying while typing a numerical street address (because while they are typing it, we assume they're looking by stop ID).  However, if someone's smart enough to know the street address, they'll probably be cool with adding one more character, which will return their desired stop.

Phew.  What do you think?

**Comment by akaplo** [comment]
Ionic 2 will make this stupidly easy.

**Comment by akaplo** [comment]
Enough.  Let's fix or close by RC-2.

**Comment by sherson** [comment]
This is mostly all set, but we should prioritize `RouteAbbreviation` over `LongName` over `RouteId`.

**Comment by akaplo** [comment]
Previous comment being taken care of by #235.

**Comment by akaplo** [comment]
Closing.
